docs/search-views/semantic-search/queries/similar.md

```diff
@@ -1,16 +1,82 @@
-Similarity search C8QL functions
-
-Similar(view_name, search_vector, [limit=100]), .e.g
-return similar("hnsw_cosine", [30, 60, 60, 20])
-return similar("hnsw_cosine", [30, 60, 60, 20], 5)
-
-Similar_to_array(search_vector, input_vectors, distance type, [limit=100]), e.g.
-return similar_to_array([100, 200, 150, 100], [[30,50,60,70],[10,20,15,10],[4,5,6,7]], \"cosine\")
-return similar_to_array([100, 200, 150, 100], [[30,50,60,70],[10,20,15,10],[4,5,6,7]], \"l2\", 3)
-Distance types can be “l2”, “ip” and “cosine” (same values as used for view creation)
-
-Support following Similarity functions in C8QL
-a. SIMILAR(queryVector, limit=100) → Array[(_key, score)]
-i . Finds documents most similar to the input vector. Returns a list of (id, score) sorted by highest score.
-b. SIMILAR(queryVector, Array[dataVectors], distance_type, limit=100) → Array[(_key, score)]
-i . Computes the similarity between input vector and array of data vectors. Returns a list of (id, score) sorted by highest score, where id is the index in data.
\ No newline at end of file
+---
+title: Similar Search Queries
+---
+
+These functions are valuable in finding the similarity between vectors, and they offer different options for comparing against a specific view or a set of vectors with the specified distance type.
+
+## SIMILAR()
+
+Search for documents most similar to the input vector.
+
+### SIMILAR Syntax
+
+- **For a specific view:**
+  `SIMILAR(view_name, search_vector, [limit=100])`
+
+### Parameters
+
+| Key            | Type         | Description                   |
+|----------------|--------------|-------------------------------|
+| view_name      | string       | Name of the view used for similarity search.                                |
+| search_vector  | array        | The input vector to search for.                                             |
+| limit          | number       | (Optional) The maximum number of results to return. Default is `100`.       |
+
+### SIMILAR Examples
+
+The examples below illustrate different use cases of the `SIMILAR()` function, demonstrating how it can be applied to find similarities against a specified view or a set of vectors using different distance types and limits.
+
+#### Example 1
+
+```c8ql
+RETURN SIMILAR("hnsw_cosine", [30, 60, 60, 20])
+```
+
+This query searches for documents most similar to the input vector `[30, 60, 60, 20]` in the view named `"hnsw_cosine"`. The result will include up to 100 matches, as the limit is not specified and defaults to 100.
+
+#### Example 2
+
+```c8ql
+RETURN SIMILAR("hnsw_cosine", [30, 60, 60, 20], 5)
+```
+
+This query is similar to the first but includes a limit of 5, meaning it will return up to 5 documents most similar to the input vector.
+
+## SIMILAR_TO_ARRAY()
+
+Compute the similarity between an input vector and an array of data vectors.
+
+### SIMILAR_TO_ARRAY Syntax
+
+`SIMILAR_TO_ARRAY(search_vector, input_vectors, distance_type, [limit=100])`
```

Comment:
both these syntaxes look same.

Reply:
Which should I use? I thought input_vectors and Array[dataVectors] might be different options or something.

Reply:
nvm, you answered below